### PR TITLE
OTWO-6666 Project Trend charts

### PR DESCRIPTION
**Pull request by Niharika1117**

### CODE REVIEW

File: app/assets/javascripts/project.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+var ProjectDashboard = {
+  init: function() {
+    $("#one_year_project, #six_months_project, #three_months_monthly_project, #six_months_monthly_project, #one_year_monthly_project").hide();
+    $('input[name="options_project"]').change( function() {
+      console.log("hiiiiii");
```
Comment (sig-krasnick) [inline app/assets/javascripts/project.js:5]: remove console message

File: app/assets/javascripts/project.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      ProjectDashboard.update_chart($(this).val());
+    })
+    $('input[name="radio_project"]').change( function() {
+      ProjectDashboard.update_chart($(this).val());
+    })
+  },
+  update_chart: function(months) {
+    if(months == '3' && $('input[name="radio_project"]').filter(':checked').val() == 'weekly') {
```
Comment (sig-krasnick) [inline app/assets/javascripts/project.js:13]: This function is too long and repetitive in general.  It should be broken into at least 2 methods.  First to determine the selection and the second to handle the showing and hiding of components.  Pass whatever you need for the appropriate show into this new method

File: app/decorators/oh_admin/project_chart.rb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+  private
+
+  def set_series_data(chart)
+    chart['series'][0][:data] = @analyzed.values
+    chart['series'][1][:data] = @non_analyzed.values
+    chart['series'][2][:data] = @total_count.flatten
+  end
+
+  def account_data
```
Comment (sig-krasnick) [inline app/decorators/oh_admin/project_chart.rb:29]: This should be changed to project_data

File: app/decorators/oh_admin/project_chart.rb [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    else
+      Project.left_joins(:best_analysis).group('date(projects.created_at)')
+             .where('projects.created_at >= ? AND projects.created_at <= ? AND projects.best_analysis_id is NULL',
+                    @from, @to).references(:best_analysis).count
+    end
+  end
+
+  def total_projects(date)
+    if @filter == 'monthly'
+      Project.left_joins(:best_analysis).group("DATE_TRUNC('month', projects.created_at)")
```
Comment (sig-krasnick) [inline app/decorators/oh_admin/project_chart.rb:102]: We shouldn't need to left join with analysis here to get the total projects

**Review by sig-krasnick — CHANGES_REQUESTED** [review]

**Comment by Niharika1117** [comment]
Retest this please

File: app/assets/javascripts/project.js [hunk 5]
```diff
@@ -1,38 +1,24 @@
...
-    }
-    else if(months == '6' && filter == 'monthly') {
-      $("#six_months_monthly_project").show();
-      $("#three_months_project, #six_months_project, #one_year_project, #three_months_monthly_project, #one_year_monthly_project").hide();
-    }
-    else if(months == '12' && filter == 'monthly') {
-      $("#one_year_monthly_project").show();
-      $("#three_months_project, #six_months_project, #one_year_project, #three_months_monthly_project, #six_months_monthly_project").hide();
+    else if(filter == 'weekly') {
+      var contClass = $('input[name="project_options"]').filter(':checked').data('div');
```
Comment (sig-krasnick) [inline app/assets/javascripts/project.js:20]: This line can be called once and passed into the function.  Otherwise looks good

Comment (Niharika1117) [reply app/assets/javascripts/project.js:20]: I tried calling that the line at once, But problem is button checked value is changing dynamically and when we click on button change function is getting called.

So, both the change function I need to call that line again.

**Review by sig-krasnick — APPROVED** [review]

**Pull request merged**